src/main/java/tutorlink/command/ListComponentCommand.java

```diff
@@ -23,6 +24,9 @@ public class ListComponentCommand extends Command {
      */
     @Override
     public CommandResult execute(AppState appState, HashMap<String, String> parameters) {
+        if(appState.components.size() <= 0) {
```

Comment:
It is good that you are adding an output message if there are no components added.